neqo-http3/src/recv_message.rs

```diff
@@ -368,6 +400,7 @@ impl RecvMessage {
                     ":scheme" => PSEUDO_HEADER_SCHEME,
                     ":authority" => PSEUDO_HEADER_AUTHORITY,
                     ":path" => PSEUDO_HEADER_PATH,
+                    ":protocol" => PSEUDO_HEADER_PROTOCOL,
```

Comment:
We should also check that this does not appear on anything other CONNECT.  Down below vvv in headers_valid, which appears to need some validation.